src/main/java/Parser.java

```diff
@@ -0,0 +1,111 @@
+public class Parser{
+    
+    private Storage storage;
+    private TaskList taskList;
+
+    public Parser(Storage storage, TaskList taskList){
+        this.storage = storage;
+        this.taskList = taskList;
+
+    }
+
+    public boolean parse (String line){
+
+        if(line.equals("bye")){
+            System.out.println("Bye. Hope to see you again soon!");
+            return false;
+        }else if(line.equals("list")){
+            taskList.printTaskList();
+            return true;
+        }else{ 
+            String inputnoun = line.split(" ")[0];
+
+            try{
+                if((!inputnoun.equals("done")) &&
+                (!inputnoun.equals("todo")) &&
+                (!inputnoun.equals("deadline")) &&
+                (!inputnoun.equals("event")) &&
+                (!inputnoun.equals("delete")) ){
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-("); 
+
+                }else if((line.split(" ", 2)).length < 2){
+                    throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                }else{
+    
+                    String taskName = line.split(" ", 2) [1];
+                    
+                    if(inputnoun.equals("done")){
+                        String stringFromUser = line.replaceAll("\\D+","");
+                        Integer indexFromUser = Integer.parseInt(stringFromUser);
+                        taskList.doneTask(indexFromUser - 1);
+                        storage.save();
+                            
+                    }else if (inputnoun.equals("todo")){
+                        ToDoTask newTask = new ToDoTask (false, taskName);
+                        taskList.add(newTask);
+                        storage.save();
+
+                    }else if (inputnoun.equals("deadline")){
```

Comment:
Perhaps you can consider placing the logic of the Date object creation, along with the associated error handling, within the constructor of the DeadlinesTask object. This way, your code for Parser can be cleaner.